### ISSUE TITLE
containsExactly behaves unexpectedly with double arrays

### ISSUE THREAD

**Issue by johnzeringue** [joel-costigliola/assertj-core#771]
#### Summary

When using `containsExactly` with double arrays and a given precision, assertions unexpectedly produce strange failures under `3.5.2`.

#### Example

```java
assertThat(new double[] {-1.71, -1.51})
        .containsExactly(new double[] {-1.7, -1.6}, within(0.1));
```

fails with

```
java.lang.AssertionError: 
Expecting:
  <[-1.71, -1.51]>
to contain exactly (and in same order):
  <[-1.7, -1.6]>
but some elements were not found:
  <[]>
and others were not expected:
  <[-1.51]>
when comparing values using 'double comparator at precision 0.1'
```

Instead, I would expect no failure, as `-1.71` is within `0.1` of `-1.7` and `-1.51` is within `0.1` of `-1.6`.

#### Java 8 specific ?

* YES : create Pull Request from the `master` branch
* NO  : create Pull Request from the `2.x` branch

**Comment by johnzeringue** [comment]
It seems like the issue is that AssertJ is deduping the expected array (note that `-1.7` is within `0.1` of `-1.6`). I'm not sure why this was originally done (the deduping code looks to predate this form of `containsExactly`), but it's clearly not valid when equality isn't transitive, as is the case here.

**Comment by joel-costigliola** [comment]
Thanks for reporting this bug, will be fixed in 2.6.0/3.6.0
